src/services/services.ts

```diff
@@ -1086,6 +1088,11 @@ namespace ts {
         newText: string;
     }
 
+    export interface TextInsertion {
+        newText: string;
+        offsetInNewText: number;

```

Comment:
offset of what?  "caretOffset" seems liek a reasonable name.  With a comment explaining that it's the offset in 'newText' where the caret shoudl be placed.